### PR TITLE
Allow different rules and action for delete user

### PR DESCRIPTION
**Pull request by jyeshe**
### Description

This PR changes the rendering of the `ProfileLive.Edit` and `UserLive.Index` (superuser) liveviews to allow them to be customized on three aspects when deleting a user:

- UI rendering
- Validation rules
- Action processing

Its strategy was similar to the Projects customization, where the liveview is injected on the router.

The change consists on making the respective `edit.html.heex` and `index.html.heex` to rely completely on components so that the Saas page can reuse them and implement different rules on the liveview modules. 

EDIT: As suggested by the LV docs, the metadata `live/4` opt can be used to customize a LiveView which may be invoked from different routes.

Refs #2500 

### Validation steps

`ProfileLive.Edit`:

0. Login with a new user (without project activity)
1. Navigate User profile and click on Delete account button at the bottom of the page
2. A popup shows a message: "Your account and credential data ..."
3. Type the user's email, confirm and the user shall be logged out. On an attempt to login it shows:
image

`UserLive.Index`:

1. Login with a superuser, enter on Admin Menu and click on Users menu
2. Click on the button to delete the Viewer user
A popup shows a message: "This user's account and credential ..."
3. Type viewer@openfn.org, confirm and it shall show the user as scheduled for deletion:
image
5. Repeat the operation for the same user and the Viewer user is deleted (doesn't appear anymore on the table).

### Additional notes for the reviewer
 
As a helper note on the refactoring, `LightningWeb.ProfileLive.MfaComponent` and `LightningWeb.ProfileLive.GithubComponent` aren't directly related to Delete User and wouldn't need to be extracted from the `ProfileLive.FormComponent`. However they are not part of the `<form>` and their actions/events are handled by their own components themselves (as expected) justifying the extraction to the same level as the `ProfileLive.FormComponent` (and not inside).  

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [x] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [ ] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

### Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: lib/lightning_web/live/profile_live/common_components.ex [hunk 1]
```diff
@@ -0,0 +1,90 @@
+defmodule LightningWeb.ProfileLive.CommonComponents do
```
Comment (elias-ba) [inline lib/lightning_web/live/profile_live/common_components.ex:1]: @jyeshe `CommonComponents` is really a great name and I like it. But we usually name these modules as `Common` or `Components`, I am wondering if we shouldn't keep that naming for consistency ?

File: lib/lightning_web/live/profile_live/common_components.ex [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      />
+      <.live_component
+        module={LightningWeb.ProfileLive.GithubComponent}
+        id={"#{@current_user.id}_github_section"}
+        user={@current_user}
+      />
+      <.delete_user_card url={@delete_user_url} />
+    </div>
+    """
+  end
```
Comment (elias-ba) [inline lib/lightning_web/live/profile_live/common_components.ex:67]: Nice

File: lib/lightning_web/live/user_live/common_components.ex [hunk 3]
```diff
@@ -0,0 +1,113 @@
+defmodule LightningWeb.UserLive.CommonComponents do
```
Comment (elias-ba) [inline lib/lightning_web/live/user_live/common_components.ex:1]: Same question for the naming here ? Again, I really think that `CommonComponents` is a great name, I am just wondering about module naming consistency.

File: lib/lightning_web/live/user_live/common_components.ex [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  use LightningWeb, :component
+
+  import PetalComponents.Table
+
+  attr :socket, :map, required: true
+  attr :users, :list, required: true
+  attr :live_action, :atom, required: true
+
+  attr :user_deletion_modal, :atom,
```
Comment (elias-ba) [inline lib/lightning_web/live/user_live/common_components.ex:10]: `:atom` ? You maybe mean `:any` (or `:map`) ?

File: lib/lightning_web/live/user_live/index.ex [hunk 5]
```diff
@@ -17,10 +17,16 @@ defmodule LightningWeb.UserLive.Index do
       |> Permissions.can?(:access_admin_space, socket.assigns.current_user, {})
 
     if can_access_admin_space do
-      {:ok,
-       assign(socket, :users, list_users())
-       |> assign(:active_menu_item, :users),
-       layout: {LightningWeb.Layouts, :settings}}
+      modal = Map.get(socket, :private, %{})[:delete_modal]
```
Comment (elias-ba) [inline lib/lightning_web/live/user_live/index.ex:20]: Nice ❤️

**Review by elias-ba — APPROVED** [review]
Hey @jyeshe, this is really a great piece of work. I learnt a lot while reviewing it, thank you. I left few comments (my reactions and curiosity questions), I would appreciate to hear your thoughts on em.

File: lib/lightning_web/live/user_live/common_components.ex (hunk 4 above)
Comment (jyeshe) [reply lib/lightning_web/live/user_live/common_components.ex:10]: It's the module name _()_

File: lib/lightning_web/live/user_live/common_components.ex (hunk 3 above)
Comment (jyeshe) [reply lib/lightning_web/live/user_live/common_components.ex]: done

File: lib/lightning_web/live/profile_live/common_components.ex (hunk 1 above)
Comment (jyeshe) [reply lib/lightning_web/live/profile_live/common_components.ex]: done

**Comment by jyeshe** [comment]
Changed to use metadata as the private works only on the connection.

File: lib/lightning_web/live/user_live/index.ex [hunk 6]
```diff
@@ -36,13 +39,13 @@ defmodule LightningWeb.UserLive.Index do
...
     |> assign(:page_title, "Users")
-    |> assign(:user, nil)
+    |> assign(:delete_user, nil)
   end
 
   defp apply_action(socket, :delete, %{"id" => id}) do
     socket
     |> assign(:page_title, "Users")
-    |> assign(:user, Accounts.get_user!(id))
+    |> assign(:delete_user, Accounts.get_user!(id))
```
Comment (stuartc) [inline lib/lightning_web/live/user_live/index.ex:48]: Why the change? It's called `user` downstream in the table component as well. I could imagine it's because we're wanting to signify that this user is selected for deletion, but the component doesn't really know about that, it's just a 'selected user' in my mind and is only deletable when paired with an action/event.

**Review by stuartc — APPROVED** [review]

File: lib/lightning_web/live/user_live/index.ex (hunk 6 above)
Comment (jyeshe) [reply lib/lightning_web/live/user_live/index.ex:48]: The renaming was for three reasons:
- It's not used anymore directly neither on the LV nor on the heex, it's passed down to 3rd level component.
- Same idea as you've mentioned the selected user that will be used somewhere for some operation
- Given the 2 above, to make it easier to search and find it quickly:
image

File: lib/lightning_web/live/components/user_deletion_modal.ex [hunk 7]
```diff
@@ -106,7 +107,7 @@ defmodule LightningWeb.Components.UserDeletionModal do
         </:title>
         <div class="">
           <p class="text-sm text-gray-500">
-            This user cannot be deleted until their auditable activities have also been purged.
+            <%= if @is_superuser_menu, do: "This user", else: "Your account" %> cannot be deleted until their auditable activities have also been purged.
```
[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/lightning_web/live/components/user_deletion_modal.ex:110]: The condition you propose is valid now that we've blocked the deletion of super users and also referring to the superuser on the 3rd person, when deleting his/her own account wouldn't harm much as well. The main reason was to make the condition as clearer as possible.

[the remark this answers is not in the document]
Comment (stuartc) [reply lib/lightning_web/live/components/user_deletion_modal.ex:110]: But the 3rd and 1st person perspective can be determined by the current user and the selected user, I'm not seeing a good enough reason to bring the layout into the mix when we have the current user available everywhere.

If the `@is_superuser_menu` option is purely for the copy, I would suggest an attr that controls it lives outside, like `is_current_user` or something. Components really should avoid knowing about the place they are mounted from.

[the remark this answers is not in the document]
Comment (jyeshe) [reply lib/lightning_web/live/components/user_deletion_modal.ex:110]: I get your point of sufficient existing conditions but it is not rare that find components that behave differently according to a context.
On the other hand, `is_current_user` is very straightforward and clear as well. Thanks
